### ISSUE TITLE
DS4Windows Static Constant Beep noise coming from Speaker on Copy/Fake PS4 Controller

### ISSUE THREAD

**Issue by dissidius** [Ryochan7/DS4Windows#2493]
Hello Legend!

Can you please fix this issue or make another build for us who don't have original Controllers ?

Here you can find the thread on reddit where i fixed this issue in version 3.0.18:

https://www.reddit.com/r/DS4Windows/comments/swb6sc/fakecopy_ps4_controller_speaker_constant_beep/

Basically the situation is this for our controllers:

"FOUND the "Cuplrit"! It's the changes introduced in Version 2.1.1
"!Changed type of output report transmitted to DS4 in BT. Now use report type 0x15 (previously 0x11)!
!Added extra data transmission flag to output report!
Changed thread affinity of hidDevice object. Link with initial high priority thread
Use a Dispatcher instance for managing the OutputSlotManager rather than Tasks
Changed how a running output slot queue is specified
Initial mouse offset refactors. Base mouse offset on percentage of full speed. Defaults to 0.500
Make button mouse offset a profile setting
Set MinWidth and MinHeight values for the main window. Contribution by dondrakon
Group some button mouse fields into a class. Reduces getter calls
Show minimum effective mouse speed in GUI
Put OS Info in log. Will be useful for debugging
Removed chance for a dead lock when stopping input controllers. Luckily I have never encountered it
Fixed a bug where a shiftKey macro accidentally saved HoldMacro and Macro status values. Contribution by mika-n
Fixed debug x86 project target by enabling the Allow Unsafe Code project setting. Contribution by mika-n""

My Custom build also available in the reddit thread where i reverted ONLY those changes above and fixed this issue is this:
https://drive.google.com/file/d/1q1TpGAcWdxsZzyIrlEeWH8C9rWHNiVZh/view

My method was very dirty as i am no Developer just used Notepad++ Compare Plugin and removed ONLY the changes present above

Would you mind giving us another more clean solution ?

Willing for any logs tests of course!

Cheers and thanks for everything

**Comment by Ryochan7** [comment]
So the main change that would matter would be the following

```
"!Changed type of output report transmitted to DS4 in BT. Now use report type 0x15 (previously 0x11)!
```

The output report started being used around the time when BT audio experiments started. Pretty sure I remember input performance being improved when moving to using output report 0x15. I could test it with output report 0x11 again out of curiosity but I doubt I would migrate back.

Besides that, maybe I can remove the Speaker output feature flag from the output report as it is still being used since I didn't want to change additional flags with version 3.1 that did not affect my experience. With version 3.1, the audio feature flags for the headphone volume were disabled which finally removed the DS4Audio class workaround for restoring the headphone volume based on what is set in Windows; now DS4Windows doesn't have to edit the headphone volume at all.

**Comment by dissidius** [comment]
> So the main change that would matter would be the following
> 
> ```
> "!Changed type of output report transmitted to DS4 in BT. Now use report type 0x15 (previously 0x11)!
> ```
> 
> The output report started being used around the time when BT audio experiments started. Pretty sure I remember input performance being improved when moving to using output report 0x15. I could test it with output report 0x11 again out of curiosity but I doubt I would migrate back.
> 
> Besides that, maybe I can remove the Speaker output feature flag from the output report as it is still being used since I didn't want to change additional flags with version 3.1 that did not affect my experience. With version 3.1, the audio feature flags for the headphone volume were disabled which finally removed the DS4Audio class workaround for restoring the headphone volume based on what is set in Windows; now DS4Windows doesn't have to edit the headphone volume at all.

I'm willing to test whatever comes into your mind as best possible solution!

Cheers and thanks

**Comment by Ryochan7** [comment]
Finally got around testing my old findings and they hold up. The DS4 does behave differently and is more responsive when using output report 0x15 rather than 0x11.

**Comment by dissidius** [comment]
> Finally got around to testing my old findings and they hold up. The DS4 does behave differently and is more responsive when using output report 0x15 rather than 0x11.

What does that mean ?

Will you make an additional build with 0x11 or can you find some other fix for our issue 😅

**Comment by dissidius** [comment]
Hello Legend!

Just tried version 3.1.2 and the issue is still present... same beep same frequency.

Disabling "Enable Output Data to DS4" as it did with other previous versions of DS4Winodws makes the speaker beep stop but of course i lose functionality.

I don't remember very well but for me on my custom version i believe the only thing that fixed the issue was switching from 0x15 to 0x11, nothing else {that i could think of abviously} worked. 

Thanks for trying to fix it by the way! Hoping you could think of something else!

Log appearing in App in version 3.1.2:

08/09/2022 00:01:41: Starting...
08/09/2022 00:01:41: Using output KB+M handler: SendInput
08/09/2022 00:01:41: Connection to ViGEmBus 1.17.333.0 established
08/09/2022 00:01:41: Searching for controllers...
08/09/2022 00:01:41: Using Shared Mode
08/09/2022 00:01:41: Found Controller: 98:B6:E9:8C:93:1C (BT) (DS4 v.2)
08/09/2022 00:01:41: Plugging in virtual X360 Controller
08/09/2022 00:01:41: Associate X360 Controller in slot #1 for input DS4 v.2 controller #1
08/09/2022 00:01:41: Controller 1 is using Profile "Default". Battery level is 62%

**Comment by dissidius** [comment]
> Finally got around to testing my old findings and they hold up. The DS4 does behave differently and is more responsive when using output report 0x15 rather than 0x11.

Hey Legend! @Ryochan7 

You closed the issue but the issue is still present even on latest DS4Windows 3.1.3

Cheers

**Comment by Ryochan7** [comment]
Took a bit to find this post. Spent some time last night screwing around and seemed to find a better configuration that actually uses output report 0x11. The key to improving performance was changing the array length to use the length used for the 0x15 output report type (334 bytes); probably has something to do with Windows behavior in the background. CRC-32 calculations had to be changed to take the actual report payload length into account rather than rely on the array length since they will now differ. 

d8c169abb0b77ac9c08ff2010e43c2b5c124e8ea

At this rate, I don't see BT audio output support ever being added so using output report 0x15 is not really necessary. Since performance seems to be better with the changes made, output report 0x11 will likely be used going forward. Even if support were ever figured out, the output report type sent could change as needed when sending an audio packet.

From the other referenced issue, I also stripped out the 0xA0 byte from the final BT output payload. Does not seem to be needed after all and it does not affect performance. I believe that has been there since the Jays2Kings days and I did not have a reason to remove it. If it would help with issues regarding copycat controllers then I guess I can leave that byte out.

Going to re-open this issue until the next DS4Windows release.
